core-web/apps/dotcms-ui/src/app/portlets/dot-categories/dot-categories-permissions/dot-categories-permissions.component.spec.ts

```diff
@@ -0,0 +1,55 @@
+import { Component, DebugElement, ElementRef, Input, SimpleChange, ViewChild } from '@angular/core';
+import { ComponentFixture, ComponentFixtureAutoDetect, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { DotCategoriesPermissionsComponent } from './dot-categories-permissions.component';
+import { DotPortletBaseModule } from '@components/dot-portlet-base/dot-portlet-base.module';
+
+@Component({
+    selector: 'dot-iframe',
+    template: ''
+})
+export class IframeMockComponent {
+    @Input() src: string;
+    @ViewChild('iframeElement') iframeElement: ElementRef;
+}
+
+describe('CategoriesPermissionsComponent', () => {
+    let component: DotCategoriesPermissionsComponent;
+    let fixture: ComponentFixture<DotCategoriesPermissionsComponent>;
+    let de: DebugElement;
+    beforeEach(async () => {
+        await TestBed.configureTestingModule({
+            declarations: [DotCategoriesPermissionsComponent, IframeMockComponent],
+            imports: [DotPortletBaseModule],
+            providers: [{ provide: ComponentFixtureAutoDetect, useValue: true }]
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(DotCategoriesPermissionsComponent);
+        component = fixture.componentInstance;
+        de = fixture.debugElement;
+        component.categoryId = '123';
+        fixture.detectChanges();
+    });
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
+    });
+
+    describe('permissions', () => {
+        beforeEach(() => {
+            component.categoryId = '123';
+            component.ngOnChanges({
+                categoryId: new SimpleChange(null, component.categoryId, true)
+            });
```

Comment:
You can't do this manually, you have to leave angular to do this, you need to update the input and use fixture.detechChanges